<robot name="robot">
  <link name="base_footprint"></link>
  <joint name="base_joint" type="fixed">
    <parent link="base_footprint" />
    <child link="base_link" />
    <origin xyz="0 0 0.0004048057655643422" rpy="0 0 0" />
  </joint>
  <link name="base_link">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.5 0.5 0.01" />
      </geometry>
      <material name="base_link-material">
        <color rgba="0 0.6038273388475408 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.5 0.5 0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.17" ixy="0" ixz="0" iyy="0.17" iyz="0" izz="0.05" />
    </inertial>
  </link>
  <joint name="Axis_0_Joint" type="revolute">
    <parent link="base_link" />
    <child link="Axis_0" />
    <origin xyz="0 0 0.035" rpy="0 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-2.5" upper="2.5" velocity="0.5"/>  </joint>
  <link name="Axis_0">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.15" length="0.059" />
      </geometry>
      <material name="Axis_0-material">
        <color rgba="0.3515325994898463 0.4735314961384573 0.9301108583738498 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.15" length="0.059" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_1_Joint" type="revolute">
    <parent link="Axis_0" />
    <child link="Axis_1" />
    <origin xyz="0 0 0.11189588647115647" rpy="1.5707963267948963 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.7" upper="1.7" velocity="0.5"/>  </joint>
  <link name="Axis_1">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.082" length="0.1" />
      </geometry>
      <material name="Axis_1-material">
        <color rgba="0.14702726648767014 0.14126329113044458 0.7304607400847158 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.082" length="0.1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_1_to_Segment_1" type="fixed">
    <parent link="Axis_1" />
    <child link="Segment_1" />
    <origin xyz="0 0.2350831500270899 0" rpy="-1.5707963267948963 0 0" />
  </joint>
  <link name="Segment_1">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.025" length="0.473" />
      </geometry>
      <material name="Segment_1-material">
        <color rgba="0.09084171117479915 0.3231432091022285 0.1844749944900301 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.025" length="0.473" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_2_Joint" type="revolute">
    <parent link="Segment_1" />
    <child link="Axis_2" />
    <origin xyz="0 -5.219894517229704e-17 0.2637568842473722" rpy="1.5707963267948963 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.6" upper="1.6" velocity="0.5"/>  </joint>
  <link name="Axis_2">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.055" length="0.1" />
      </geometry>
      <material name="Axis_2-material">
        <color rgba="0.14702726648767014 0.14126329113044458 0.7304607400847158 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.055" length="0.1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_2_to_Segment_2" type="fixed">
    <parent link="Axis_2" />
    <child link="Segment_2" />
    <origin xyz="0 0.19535682173790003 0" rpy="-1.5707963267948963 0 0" />
  </joint>
  <link name="Segment_2">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.025" length="0.393" />
      </geometry>
      <material name="Segment_2-material">
        <color rgba="0.09084171117479915 0.3231432091022285 0.1844749944900301 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.025" length="0.393" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_3_Joint" type="revolute">
    <parent link="Segment_2" />
    <child link="Axis_3" />
    <origin xyz="0 -4.337792830220178e-17 0.199625776257357" rpy="1.5707963267948963 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.6" upper="1.6" velocity="0.5"/>  </joint>
  <link name="Axis_3">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.05" length="0.1" />
      </geometry>
      <material name="Axis_3-material">
        <color rgba="0.14702726648767014 0.14126329113044458 0.7304607400847158 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.05" length="0.1" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_3_to_Segment_3" type="fixed">
    <parent link="Axis_3" />
    <child link="Segment_3" />
    <origin xyz="0 0.06725724726912972 0" rpy="-1.5707963267948963 0 0" />
  </joint>
  <link name="Segment_3">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.025" length="0.135" />
      </geometry>
      <material name="Segment_3-material">
        <color rgba="0.09084171117479915 0.3231432091022285 0.1844749944900301 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.025" length="0.135" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Wrist_Joint" type="revolute">
    <parent link="Segment_3" />
    <child link="Axis_4" />
    <origin xyz="0 0 0.0655808825338593" rpy="0 1.5707963267948966 0" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-1.6" upper="1.6" velocity="0.5"/>  </joint>
  <link name="Axis_4">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.03" length="0.075" />
      </geometry>
      <material name="Axis_4-material">
        <color rgba="0.23455058215026167 0.9301108583738498 0.21952619971859377 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.03" length="0.075" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Continuous_Joint" type="revolute">
    <parent link="Axis_4" />
    <child link="Axis_4_C" />
    <origin xyz="0.0009533507860803557 0 0" rpy="0 -1.5707963267948966 0" />
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="Axis_4_C">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.02" length="0.01" />
      </geometry>
      <material name="Axis_4_C-material">
        <color rgba="0.006048833020386069 0.407240211891531 0.15592646369776456 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <cylinder radius="0.02" length="0.01" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.3333333333333333" ixy="0" ixz="0" iyy="0.5" iyz="0" izz="0.3333333333333333" />
    </inertial>
  </link>
  <joint name="Axis_4_C_to_Effector" type="fixed">
    <parent link="Axis_4_C" />
    <child link="Effector" />
    <origin xyz="0 0 0.06478774571448076" rpy="0 0 0" />
  </joint>
  <link name="Effector">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.05 0.01 0.135" />
      </geometry>
      <material name="Effector-material">
        <color rgba="0.2746773120495699 0.01680737574872402 0.5711248294565854 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.01 0.01 0.135" />
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="1" />
      <inertia ixx="0.16666666666666666" ixy="0" ixz="0" iyy="0.16666666666666666" iyz="0" izz="0.16666666666666666" />
    </inertial>
  </link>
</robot>